src/Features/CSharp/Portable/Diagnostics/Analyzers/CSharpPreferFrameworkTypeDiagnosticAnalyzer.cs

```diff
@@ -0,0 +1,38 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Immutable;
+using Microsoft.CodeAnalysis.CSharp.Extensions;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Diagnostics.PreferFrameworkType;
+
+namespace Microsoft.CodeAnalysis.CSharp.Diagnostics.Analyzers
+{
+    [DiagnosticAnalyzer(LanguageNames.CSharp)]
+    internal class CSharpPreferFrameworkTypeDiagnosticAnalyzer : PreferFrameworkTypeDiagnosticAnalyzerBase<SyntaxKind>
+    {
+        protected override ImmutableArray<SyntaxKind> SyntaxKindsOfInterest => 
+            ImmutableArray.Create(SyntaxKind.PredefinedType);
+
+        protected override bool IsPredefinedTypeAndReplaceableWithFrameworkType(SyntaxNode node)
+        {
+            var syntaxKind = (node as PredefinedTypeSyntax)?.Keyword.Kind();
```

Comment:
I wouldn't.  If you asked for some specific data, you should only get that data.